make/buf/scripts/fuzz.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+CDPATH="" cd "$(dirname "${0}")/../../.."
+FUZZ_DIR="${FUZZ_DIR:-$(pwd)/.tmp/gofuzz}"
+GO_FUZZ_VERSION="${GO_FUZZ_VERSION:-master}"
+
+git diff --exit-code --quiet go.mod go.sum || (echo "go.sum and go.mod must be unmodified" && exit 1)
+
+go get github.com/dvyukov/go-fuzz/go-fuzz-dep@"$GO_FUZZ_VERSION"
```

Comment:
Why is this version managed in this script rather than in the `make` target?

Reply:
GO_FUZZ_VERSION's value comes from dep_go_fuzz.mk when this script is called from the gofuzz target. [This line](https://github.com/WillAbides/buf/blob/13394c98ffee7b25c9e3aee5e7ad9a954b48eef4/make/buf/scripts/fuzz.sh#L8) may make it look like this script is managing the version, but that is just setting a default value.

I could be more clear by removing the default value and erroring when no value is set.

Reply:
Sorry, I was being unclear in my initial message - I mean, why isn't this binary just installed together with `go-fuzz` and `go-fuzz-build`?

Reply:
I see what you mean now. This doesn't install a binary. It just adds "github.com/dvyukov/go-fuzz" to go.mod and go.sum so that go-fuzz-build can build an instrumented binary.

Now that you bring it up, we could have this script install go-fuzz and go-fuzz-build as well so that it's all handled in one place...or is that what you were suggesting before?

Reply:
I think I'd prefer if we can handle it in the make targets (like `dep_gofuzz.mk`), since that's the pattern we generally follow for binary dependencies. It sounds like this isn't exactly that, but that `go-fuzz` needs `go-fuzz` in its dependency closure when building the binary. I'm not sure I understand this part, could you elaborate a little bit? It's not imported anywhere, so what does adding it to the dependency closure accomplish?

Reply:
dependencies should be exclusively installed via the makefile commands, and there should be a corresponding make target to invoke this script that depends on the installation of the dependency

Reply:
this should be removed

Reply:
go-fuzz-build loads "github.com/dvyukov/go-fuzz/go-fuzz-dep" and errors if it is missing from go.mod.  This is where it happens:

https://github.com/dvyukov/go-fuzz/blob/b1f3d6f4ef4e0fab65fa66f9191e6b115ad34f31/go-fuzz-build/main.go#L297-L313

Reply:
@bufdev 
> dependencies should be exclusively installed via the makefile commands, and there should be a corresponding make target to invoke this script that depends on the installation of the dependency

This doesn't install a dependency, or at least not a binary dependency. All it does is temporarily add "github.com/dvyukov/go-fuzz/go-fuzz-dep" to the module so that `go-fuzz-build` can use it.

Reply:
I see, that sounds like a pretty unique requirement of the fuzz program. I can't wait for Go 1.17 when we can hopefully replace it with `testing.F`!